src/d3a/d3a_core/export.py

```diff
@@ -384,7 +387,52 @@ def _plot_ess_soc_history(self, storage_list: list, subdir: str, root_name: str)
         plot_dir = os.path.join(self.plot_dir, subdir)
         mkdir_from_str(plot_dir)
         output_file = os.path.join(plot_dir, 'ess_soc_history_{}.html'.format(root_name))
-        BarGraph.plot_bar_graph(barmode, title, xtitle, ytitle, data, output_file)
+        PlotlyGraph.plot_bar_graph(barmode, title, xtitle, ytitle, data, output_file)
+
+    def supply_demand_curve(self, area, subdir):
+        """
+        Wrapper for _supply_demand_curve
+        """
+        new_subdir = os.path.join(subdir, area.slug)
+        self._supply_demand_curve(new_subdir, area)
+        for child in area.children:
+            if child.children:
+                self.supply_demand_curve(child, new_subdir)
+
+    def _supply_demand_curve(self, subdir: str, area: Area):
+
+        for past_market in area.past_markets:
+            data = list()
+            xmax = 0
+            for time_slot, supply_curve in past_market.state.cumulative_offers.items():
+                data.append(PlotlyGraph._line_plot(supply_curve, time_slot, True))
+            for time_slot, demand_curve in past_market.state.cumulative_bids.items():
+                data.append(PlotlyGraph._line_plot(demand_curve, time_slot, False))
+            for time_slot, clearing_point in past_market.state.clearing.items():
+                # clearing_point[0] --> Clearing-Rate
+                # clearing_point[1] --> Clearing-Energy
+                if len(clearing_point) != 0:
+                    data_obj = go.Scatter(x=[0, clearing_point[1]],
+                                          y=[clearing_point[0], clearing_point[0]],
+                                          mode='lines+markers',
+                                          line=dict(width=5),
+                                          name=str(time_slot) + str('Clearing-Rate'))
+                    data.append(data_obj)
+                    data_obj = go.Scatter(x=[clearing_point[1], clearing_point[1]],
+                                          y=[0, clearing_point[0]],
+                                          mode='lines+markers',
+                                          line=dict(width=5),
+                                          name=str(time_slot) + str('Clearing-Energy'))
+                    data.append(data_obj)
+                    xmax = max(xmax, clearing_point[1]) * 3
+
+            if len(data) > 0:
```

Comment:
You could move this check up to line 413 to skip the plotting of the clearing point if there is no supply and demand curve.